content-services/content-sources-backend
# Fixes 4571: list snapshots by template with repo information

## Summary
This PR adds a new endpoint for listing snapshots by template and adds 2 new fields `repository_name` and `repository_uuid` to API snapshot response (for all endpoints).
The new added fields required adding preloading to and modifying queries in other endpoints.
New endpoint had to be added, rather than modifying the existing list one, because the process of querying and filtering the snapshots is completely different.

## Testing steps
1. Create a new template with some repositories that have snapshots.
2. Query the snapshots for the template. `/templates/{uuid}/snapshots/` and verify the shown snapshots are actually closest one for the template date.
3. Check the filtering of snapshots with the query param `repository_search` and sorting by repository name (`sort_by=repository_name:asc|desc`) or by created_at.
